### PR TITLE
Annotations on IstioOperator CR

### PR DESCRIPTION
**Pull request by jdmaguire**
I'm using ArgoCD to deploy this operator. I need to add an annotation to the CR to the istio-operator chart to handle some lifecycle hooks better.

### CODE REVIEW

File: charts/istio-operator/templates/istiooperator.yaml [hunk 1]
```diff
@@ -4,6 +4,10 @@ kind: IstioOperator
 metadata:
   namespace: {{ .Values.istioNamespace }}
   name: istio-operator
+  annotations:
+    {{- if .Values.controlPlane.annotations }}
+    {{- toYaml .Values.controlPlane.annotations | nindent 4 }}
+    {{- end }}
```
Comment (stevehipwell) [inline charts/istio-operator/templates/istiooperator.yaml:10]:
```suggestion
  {{- with .Values.controlPlane.annotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```

We only want to add annotations if there are any being passed in.

**Review by stevehipwell — CHANGES_REQUESTED** [review]
Thanks for the PR @jdmaguire, this looks good but I've requested a change to how you've templated the annotations. Could you also add a new `## [UNRELEASED]` heading to _CHANGELOG.md_ and add you change to that ready for when the chart is released?

**Review by jdmaguire — COMMENTED** [review]
Release note for contolPlane.annotations

**Comment by stevehipwell** [comment]
@jdmaguire could you rebase and then I'll run the automation.

**Review by stevehipwell — APPROVED** [review]
LGTM

Thanks for the PR @jdmaguire. 👏🏻

**Pull request merged**